analyzers/tests/SonarAnalyzer.Test/TestCases/SpecifyRouteAttribute.CSharp12.cs

```diff
@@ -190,3 +190,23 @@ public class DerivedController : Controller { }
 
 [Controller]
 public class Endpoint { }
+
+public class ControllerWithInheritedRoute : ControllerWithRouteAttribute            // Compliant, attribute is inherited
+{
+    [HttpGet("Test")]
+    public string Index() => "Hi!";
+}
+
+[Route("api/[controller]")]
+public class ControllerWithRouteAttribute : Controller { }
+
+public class ControllerOverridesActionWithRoute : BaseControllerWithActionWithRoute // Noncompliant
+{
+    public override string Index() => "Hi!";                                        // Secondary
+}
+
+public class BaseControllerWithActionWithRoute : Controller                         // Noncompliant
+{
+    [HttpGet("Test")]
+    public virtual string Index() => "Hi!";                                         // Secondary
```

Comment:
Add an overload without an attribute:
```suggestion
    [HttpGet("Test")]
    public virtual string Index() => "Hi!";                                         // Secondary
    
    public virtual string Index(int id) => "Hi!";                                 // Compliant
    
    // ...
    public override string Index(int id) => "Hi!"; // Compliant
```